File: 13_Remove_Duplicates/Golang/Main.go
package main

import (
	"fmt"
)

func removeDuplicates(nums []int) int {
	var (
		element = nums[0]
		l       = 1
	)

	for i := 1; i < len(nums); i++ {
		if nums[i] != element {
			nums[l] = nums[i]
			l++
		}

		element = nums[i]
	}

	return l
}

func main() {

	x := removeDuplicates([]int{1, 1, 2})
	fmt.Println(x)
}

File: 13_Remove_Duplicates/JavaScript/Main.js
let removeDuplicates = function (nums) {
  let element = nums[0]
  let l = 1

  for (let i = 1; i < nums.length; i++) {
    if (nums[i] !== element) {
      nums[l] = nums[i]
      l++
    }

    element = nums[i]
  }

  return l
}

console.log(removeDuplicates([1, 1, 2]));
